### ISSUE TITLE
Newton: HAProxy playbook does not create config files

### ISSUE THREAD

**Issue by melissa-kam** [rcbops/rpc-openstack#1663]
##### What type of issue is this? Please tick the appropriate box.
- [x] bug - existing functionality that does not work for you as documented
- [ ] feature - new functionality that you would like to see added

##### What is the expected behavior?
The [haproxy.yml playbook](https://github.com/rcbops/rpc-openstack/blob/master/rpcd/playbooks/haproxy.yml) runs all steps successfully.

##### What is actually happening?
The ["Create haproxy service config files" task](https://github.com/openstack/openstack-ansible-haproxy_server/blob/stable/newton/tasks/haproxy_service_config.yml) is skipped.
```
TASK [haproxy_server : Create haproxy service config files] ********************
[DEPRECATION WARNING]: Skipping task due to undefined Error, in the future this
 will be a fatal error.: 'haproxy_service_configs' is undefined.
This feature 
will be removed in a future release. Deprecation warnings can be disabled by 
setting deprecation_warnings=False in ansible.cfg.
skipping: [loadbalancer1]
```

##### What were you attempting to do when you encountered this error (give as much detail as possible)? 
I ran deploy.sh with `DEPLOY_ELK=yes` and `DEPLOY_HAPROXY=yes` in a multi-node all in one environment on a public cloud server, and Kibana was not accessible. The step in deploy.sh that executes `run_ansible haproxy-install.yml` instead doesn't have this issue, but `run_ansible haproxy.yml` in deploy-rpc-playbooks.sh does.

##### How can we reproduce this issue?
```
export DEPLOY_ELK=yes
export DEPLOY_HAPROXY=yes
./rpc-openstack/scripts/deploy.sh
```
or
```
openstack-ansible haproxy.yml
```

##### How can we verify your issue is addressed?
The "Create haproxy service config files" should run without being skipped and Kibana should be accessible if haproxy.yml is run after ELK is deployed. This can be verified manually by inspecting the console output of the JJB-RPC-AIO jobs or automatically with OnMetal-All-In-One-Pipeline job in the QE Jenkins since that pipeline has Kibana tests.

##### If you are using or know of a workaround please describe the necessary steps?
Using haproxy-install.yml in openstack-ansible instead of haproxy.yml in rpc-openstack will create the config files properly.

##### What is your evaluation on the importance of this issue, and why?
Effects Kibana testing in AIO environments that use HAProxy, though we can temporarily add the workaround as an additional step to proceed with automated testing.

----------------------------------------------------------------------------------------------------------------------------
###### For each pull request, update the branch line to reference it.
- [ ] master
- [ ] mitaka-13.1
- [ ] liberty-12.2
- [ ] kilo

##### Upgrade impact. Consider the impact on moving between major, minor and patch releases.
- [ ] This change does not impact upgrades
- [ ] This change addresses a potential upgrade impact
- [ ] The upgrade impact is being tracked by another issue.

The issue tracking the upgrade impact is:

##### Testing impact
- [ ] An issue/pull-request has been create or exists to track the testing gap, and is referenced below
- [ ] This change includes additional testing
- [x] This change is tested by the existing setup

Additional testing links:

##### Release note
- [ ] This change includes a release note.
- [ ] This change does not require a release note.

##### Does your change require documentation?
- [ ] Yes, and I submitted an issue to [Docs issues](https://github.com/rackerlabs/docs-rpc/issues "Docs issues")
- [ ] Yes, and I contributed a documentation patch to [docs-rpc](https://github.com/rackerlabs/docs-rpc "docs-rpc")
- [x] No documentation required for this change

Additional documentation links (please add links if you have ticked yes):

**Comment by odyssey4me** [comment]
the current mechanism for adding the ELK haproxy configuration should be changed to use the ``haproxy_extra_services`` variable. This will be more efficient and allow the removal of the haproxy playbook in RPC. As far as I recall this functionality was also implemented in Mitaka so the patch could be backported to Mitaka too.

**Comment by alextricity25** [comment]
FYI: The fix for this will also need to be included in the mitaka-13.1 branch. This is due to the changes that were introduced here: https://review.openstack.org/#/c/404328/1
I'm also bumping the priority of this issue since it's a blocker for https://github.com/rcbops/rpc-openstack/pull/1675
